spectator-api/src/test/java/com/netflix/spectator/api/patterns/ThreadPoolMonitorTest.java

```diff
@@ -0,0 +1,280 @@
+    ThreadPoolMonitor.monitor(registry, threadPoolExecutor, THREAD_POOL_NAME);
+    PolledMeter.update(registry);
+    Id id = registry.createId(meterName).withTag(ThreadPoolMonitor.ID_TAG_NAME, THREAD_POOL_NAME);
+    return registry.get(id);
+  }
+
+  @Test
+  public void metricsAreTaggedWithProvidedThreadPoolName() {
+    Meter meter = getMeter(ThreadPoolMonitor.MAX_THREADS);
+
+    Iterable<Measurement> measurements = meter.measure();
+    Iterator<Measurement> measurementIterator = measurements.iterator();
+    assertTrue(measurementIterator.hasNext());
+
+    Iterator<Tag> tags = measurementIterator.next().id().tags().iterator();
+    assertTrue(tags.hasNext());
+    assertEquals(getClass().getSimpleName(), tags.next().value());
+  }
+
+  @Test
+  public void threadPoolMonitorHasTaskCountMeter() {
+    assertNotNull(getMeter(ThreadPoolMonitor.TASK_COUNT));
+  }
+
+  @Test
+  public void threadPoolMonitorHasCompletedTaskCountMeter() {
+    assertNotNull(getMeter(ThreadPoolMonitor.COMPLETED_TASK_COUNT));
+  }
+
+  @Test
+  public void threadPoolMonitorHasCurrentThreadsBusyMeter() {
+    assertNotNull(getMeter(ThreadPoolMonitor.CURRENT_THREADS_BUSY));
+  }
+
+  @Test
+  public void threadPoolMonitorHasMaxThreadsMeter() {
+    assertNotNull(getMeter(ThreadPoolMonitor.MAX_THREADS));
+  }
+
+  @Test
+  public void threadPoolMonitorHasPoolSizeMeter() {
+    assertNotNull(getMeter(ThreadPoolMonitor.POOL_SIZE));
+  }
+
+  @Test
+  public void threadPoolMonitorHasCorePoolSizeMeter() {
+    assertNotNull(getMeter(ThreadPoolMonitor.CORE_POOL_SIZE));
+  }
+
+  @Test
+  public void threadPoolMonitorHasQueueSizeMeter() {
+    assertNotNull(getMeter(ThreadPoolMonitor.QUEUE_SIZE));
+  }
+
+  @Test
+  public void maxThreadsUpdatesWhenRegistryIsUpdated() {
+    Meter meter = getMeter(ThreadPoolMonitor.MAX_THREADS);
+    Iterator<Measurement> measurements = meter.measure().iterator();
+    assertEquals(10.0, measurements.next().value(), 0.0);
+
+    threadPoolExecutor.setMaximumPoolSize(42);
+    PolledMeter.update(registry);
+    measurements = meter.measure().iterator();
+    assertEquals(42.0, measurements.next().value(), 0.0);
+  }
+
+  @Test
+  public void corePoolSizeUpdatesWhenRegistryIsUpdated() {
+    Meter meter = getMeter(ThreadPoolMonitor.CORE_POOL_SIZE);
+    Iterator<Measurement> measurements = meter.measure().iterator();
+    assertEquals(3.0, measurements.next().value(), 0.0);
+
+    threadPoolExecutor.setCorePoolSize(42);
+    PolledMeter.update(registry);
+    measurements = meter.measure().iterator();
+    assertEquals(42.0, measurements.next().value(), 0.0);
+  }
+
+  @Test
+  public void taskCountUpdates() throws InterruptedException {
+    Meter meter = getMeter(ThreadPoolMonitor.TASK_COUNT);
+    Iterator<Measurement> measurements = meter.measure().iterator();
+    assertEquals(0.0, measurements.next().value(), 0.0);
+
+    Cancellable command = new Cancellable();
+    threadPoolExecutor.execute(command);
+
+    Thread.sleep(60);
```

Comment:
I would prefer to avoid the use of sleep in tests. In my experience that tends to make them extremely brittle. A latch should work here:

```java
final CountDownLatch latch = new CountDownLatch(1);
threadPoolExecutor.execute(latch::countDown);
latch.await(5, TimeUnit.SECONDS);
```

Reply:
Reworked to use CountDownLatch.